sql/item_func.h

```diff
@@ -2681,6 +2681,15 @@ class Item_func_get_lock :public Item_long_func
   { return get_item_copy<Item_func_get_lock>(thd, this); }
 };
 
+class Item_func_release_all_locks :public Item_long_func 
+{
+public:
+  Item_func_release_all_locks(THD *thd): Item_long_func(thd) { unsigned_flag=1; }
+  longlong val_int() override;
+  const char *func_name() const override { return "release_all_locks"; }
+  Item *get_copy(THD *thd) override  { return get_item_copy<Item_func_release_all_locks>(thd,this); }
+};
```

Comment:
I think this function should do the same things that Item_func_release_lock and Item_func_get_lock do:

```
  table_map used_tables() const
  {
    return used_tables_cache | RAND_TABLE_BIT;
  }
  bool const_item() const { return 0; }
  bool is_expensive() { return 1; }
  bool check_vcol_func_processor(void *arg)
  {
    return mark_unsupported_function(func_name(), "()", arg, VCOL_IMPOSSIBLE);
  }
```

To avoid code duplication, please introduce an abstract class, say Item_func_long_lock, move the mentioned methods there, then derive Item_func_get_lock, Item_func_release_lock, Item_func_release_all_locks from the new abstract class.

Reply:
Alexander, would you explain what engine ask function all this questions? 
bool const_item() const { return 0; }
bool is_expensive() { return 1; }
bool check_vcol_func_processor(void *arg)
And why do we need method table_map used_tables if patch seems to work fine without it?